### PR TITLE
#40 testing

### PR DESCRIPTION
**Pull request by MMCQ33**
Added More testing

### CODE REVIEW

File: Hivemind/Assets/BaseBuildingTestScene.unity [hunk 1]
```diff
@@ -0,0 +1,6122 @@
+%YAML 1.1
```
Comment (ErikLouwerse) [inline Hivemind/Assets/BaseBuildingTestScene.unity]: Please put this in the scenes folder

File: Hivemind/Assets/Scenes/BaseBuildingTestScene 1.meta [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scenes/BaseBuildingTestScene 1.meta]: What meta file is this for?

File: Hivemind/Assets/Scripts/Ant.cs [hunk 3]
```diff
@@ -183,6 +198,11 @@ private void ChangeGroupID(object sender, GroupIdChangedEventArgs e)
         }
     }
 
+    public List<IMind> GetMinds()
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Ant.cs:248]: It's already public.
Please remove or make public set function and private the variable

File: Hivemind/Assets/Scripts/MindGroup.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ public MindGroup(GameObject UiObject)
 
     public int Count { get; private set; }
 
-    public List<IMind> Minds { get; }
+    public List<IMind> Minds { get; set; }
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/MindGroup.cs]: pls remove getset as it is useless

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 5]
```diff
@@ -238,7 +241,13 @@ private void TargetResource()
 
         if (target != null)
         {
-            if (state == State.Idle) ant.GetAgent().isStopped = false;
+            if (state == State.Idle)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: just make double if

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 6]
```diff
@@ -290,9 +302,35 @@ private void carryResource(ResourceNode resource)
     }
 
     private IEnumerator Scout()
+    {
+
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: WHITESPACE

File: Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+using UnityEngine.TestTools;
+using System.Linq;
+
+namespace Tests.PlayModeTests
+{
+    public class BaseBuildingTest
+    {
+        private GameObject gameUI;
+        private UiController uiController;
+        private UnitController unitControl;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs]: Useless

File: Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        [UnitySetUp]
+        public IEnumerator Init()
+        {
+            SceneManager.LoadScene("BaseBuildingTestScene");
+            yield return new WaitForSeconds(1f);
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs]:
```suggestion
            AsyncOperation asyncLoadLevel = SceneManager.LoadSceneAsync("BaseBuildingTestScene", LoadSceneMode.Single);
            while (!asyncLoadLevel.isDone)
            {
                Debug.Log("Loading the Scene");
                yield return null;
            }
```

File: Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+
+namespace Tests.PlayModeTests
+{
+    public class GatherResourceTest
+    {
+
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs]: WHITESPACE

File: Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        private UiController uiController;
+        private UnitController unitControl;
+        private GameObject resource;
+        private ResourceNode resourceNode;
+
+        [UnitySetUp]
+        public IEnumerator Init()
+        {
+            SceneManager.LoadScene("TestScene");
+            yield return new WaitForSeconds(1f);
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs]:
```suggestion
            AsyncOperation asyncLoadLevel = SceneManager.LoadSceneAsync("TestScene", LoadSceneMode.Single);
            while (!asyncLoadLevel.isDone)
            {
                Debug.Log("Loading the Scene");
                yield return null;
            }
```

File: Hivemind/Assets/Tests/PlayModeTests/MindBuilderTest.cs [hunk 11]
```diff
@@ -0,0 +1,118 @@
+﻿using NUnit.Framework;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Tests/PlayModeTests/MindBuilderTest.cs:1]: too much bad white space to comment

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

**Review by zoompower — APPROVED** [review]

**Review by ErikLouwerse — APPROVED** [review]

**Pull request merged**